zerver/decorator.py

```diff
@@ -74,6 +75,21 @@
 ReturnT = TypeVar("ReturnT")
 
 
+def self_hosting_management_endpoint(
+    view_func: Callable[Concatenate[HttpRequest, ParamT], HttpResponse]
+) -> Callable[Concatenate[HttpRequest, ParamT], HttpResponse]:
+    @wraps(view_func)
+    def _wrapped_view_func(
+        request: HttpRequest, /, *args: ParamT.args, **kwargs: ParamT.kwargs
+    ) -> HttpResponse:
+        subdomain = get_subdomain(request)
+        if subdomain != settings.SELF_HOSTING_MANAGEMENT_SUBDOMAIN:
+            return render(request, "404.html", status=404)
```

Comment:
Let's make this `if not settings.DEVELOPMENT or subdomain != settings.SELF_HOSTING_MANAGEMENT_SUBDOMAIN:`;

That way this code is not enabled in production until we feel good enough about its testing to feel comfortable enabling it.

Reply:
Made this tweak. Thought this leaves `remote_server_billing_entry` and `/self-hosted-billing/` endpoints not disabled. Do we want to put `if not settings.DEVELOPMENT` in their code too?

Reply:
Yeah let's do that, just so we don't need to think about the consequences of possible bugs in them.

Reply:
Cool, done